kettle-plugins/browse/src/main/java/org/pentaho/big/data/impl/browse/model/NamedClusterFile.java

```diff
@@ -71,8 +71,10 @@ public static NamedClusterFile create( String parent, FileObject fileObject ) {
     }
 
     NamedClusterFile compare = (NamedClusterFile) obj;
+    // This comparison depends on `getProvider()` to always return a hardcoded value
     return compare.getProvider().equals( getProvider() )
-      && ( ( compare.getPath() == null && getPath() == null ) || compare.getPath().equals( getPath() ) );
+      && ( ( compare.getPath() == null && getPath() == null ) || ( compare.getPath() != null && compare.getPath()
```

Comment:
No change, could also use apache's StringUtils library to replace line 75
`StringUtils.equals(compare.getPath(), getPath())`